homeassistant/components/elmax/cover.py

```diff
@@ -0,0 +1,137 @@
+        # In case the panel is offline, its status will be None. In that case, simply do nothing
+        if panel_status is None:
+            return
+
+        # Otherwise, add all the entities we found
+        entities = []
+        for cover in panel_status.covers:
+            # Skip already handled devices
+            if cover.endpoint_id in known_devices:
+                continue
+            entity = ElmaxCover(
+                panel=coordinator.panel_entry,
+                elmax_device=cover,
+                panel_version=panel_status.release,
+                coordinator=coordinator,
+            )
+            entities.append(entity)
+
+        if entities:
+            async_add_entities(entities)
+            known_devices.update([e.unique_id for e in entities])
+
+    # Register a listener for the discovery of new devices
+    config_entry.async_on_unload(coordinator.async_add_listener(_discover_new_devices))
+
+    # Immediately run a discovery, so we don't need to wait for the next update
+    _discover_new_devices()
+
+
+class ElmaxCover(ElmaxEntity, CoverEntity):
+    """Elmax Cover entity implementation."""
+
+    _attr_supported_features = (
+        CoverEntityFeature.OPEN | CoverEntityFeature.CLOSE | CoverEntityFeature.STOP
+    )
+
+    def __check_cover_status(self, status_to_check: CoverStatus) -> bool | None:
+        """Check if the current cover entity is in a specific state."""
+        state = self.coordinator.get_cover_state(self._device.endpoint_id).status
+        if state is None:
+            return None
+        return state == status_to_check
+
+    @property
+    def is_closed(self) -> bool | None:
+        """Tells if the cover is closed or not."""
+        return self.coordinator.get_cover_state(self._device.endpoint_id).position == 0
+
+    @property
+    def current_cover_position(self) -> int | None:
+        """Return current position of cover.
+
+        None is unknown, 0 is closed, 100 is fully open.
+        """
+        return self.coordinator.get_cover_state(self._device.endpoint_id).position
+
+    @property
+    def is_opening(self) -> bool | None:
+        """Tells if the cover is opening or not."""
+        return self.__check_cover_status(CoverStatus.UP)
+
+    @property
+    def is_closing(self) -> bool | None:
+        """Return if the cover is closing or not."""
+        return self.__check_cover_status(CoverStatus.DOWN)
+
+    async def async_stop_cover(self, **kwargs: Any) -> None:
+        """Stop the cover."""
+        # To stop the cover, Elmax requires us to re-issue the same command once again.
+        # To detect the current motion status, we request an immediate refresh to the coordinator
+        await self.coordinator.async_request_refresh()
+        motion_status = self.coordinator.get_cover_state(
+            self._device.endpoint_id
+        ).status
+        if motion_status is None:
+            _LOGGER.warning("Could not retrieve motion_status for elmax cover")
```

Comment:
I think maybe you should raise HomeAssistantError so the user get's feedback to UI

Reply:
Thanks for the suggestion. I've actually decided to completely remove the check as the motion_status is supposed to always be returned by the low-level API library. The underlying coordinator has been updated accordingly.

Done in [0f167c5](https://github.com/home-assistant/core/pull/79409/commits/0f167c5a0894f2b9d166047e695e0e3b76279348)